src/d3d12/mod.rs

```diff
@@ -903,13 +917,13 @@ impl Allocator {
     }
 
     /// Free a resource and its memory.
-    pub fn free_resource(&mut self, resource: Resource) -> Result<()> {
-        let _ = ManuallyDrop::into_inner(resource.resource);
-        if let Some(allocation) = resource.allocation {
+    pub fn free_resource(&mut self, mut resource: Resource) -> Result<()> {
+        let _ = resource.resource.take();
+        if let Some(allocation) = resource.allocation.take() {
```

Comment:
Never mind about this `.take()`, the field is implicitly moved and the compiler disallows you from using it entirely (instead of treating it as a possibly-`None` value:

```
borrow of partially moved value: `resource.allocation`
partial move occurs because value has type `d3d12::Allocation`, which does not implement the `Copy` trait
```

```suggestion
        if let Some(allocation) = resource.allocation {
```

Reply:
Ah, you can't because of the custom `Drop` implementation, never mind!